src/charm.py

```diff
@@ -219,27 +235,74 @@ def self_scraping_job(self):
         """The self-monitoring scrape job."""
         external_url = urlparse(self._external_url)
         metrics_path = f"{external_url.path.rstrip('/')}/metrics"
-        target = (
-            f"{external_url.hostname}{':'+str(external_url.port) if external_url.port else ''}"
-        )
+        internal_url = self._internal_url.replace("http://", "")
+        target = f"{internal_url}"
         job = {
             "metrics_path": metrics_path,
             "static_configs": [{"targets": [target]}],
         }
 
         return [job]
 
+    def _update_blackbox_config_yaml_from_relation(self, modules) -> None:
+        """Update the blackbox config yaml with modules defined in relation.
+
+        This function takes the modules from the BlackboxExporterRequirer and
+        updates the config file with the required modules.
+        """
+        config_file_data = self.container.pull(self._config_path).read()
+
+        if not config_file_data:
+            return
+
+        config_data = yaml.safe_load(config_file_data)
+
+        if "modules" not in config_data:
+            config_data["modules"] = {}
+
+        for module_name, module_data in modules.items():
+            if module_name not in config_data["modules"]:
+                config_data["modules"][module_name] = module_data
+
+        updated_config_data = yaml.safe_dump(config_data)
+        self.container.push(self._config_path, updated_config_data)
+        self.blackbox_workload.reload()
+
+    def _merge_scrape_configs(self, file_probes, relation_probes) -> list:
+        """Merge the scrape_configs from both file and relation."""
+        merged_scrape_configs = {
+            probe["job_name"]: probe for probe in file_probes.get("scrape_configs", [])
```

Comment:
Does this mean we could have a "job_name" key with value `[]`? Is that intended?